### ISSUE TITLE
Unable to execute on IOS

### ISSUE THREAD

**Issue by ashok-sl** [vitorpamplona/react-native-ble-advertiser#21]
I am stuck with the following issue. Can't even run a sample setup.
Platform - IOS
xCode Version - 12.3
Device - iPad

![Screenshot 2021-01-13 at 5 43 44 PM](https://user-images.githubusercontent.com/63572942/104451323-61a52d80-55c7-11eb-8707-cc43763fbb93.png)

**Comment by vitorpamplona** [comment]
Hum... I have never tested on an iPad. Is this a new one? Have you tried running on a simulator just to see if it goes through?

**Comment by ashok-sl** [comment]
Yes, First I tried with iPhone 11 simulator but there also the same error.

**Comment by milosh86** [comment]
Hi, I had the same issue ... try to find additional info on why it failed. I can't remember where but I found that explanation and as I remember it was because of "Privacy - Bluetooth Peripheral Usage Description" was missing in Info.plist

**Comment by vitorpamplona** [comment]
Version 0.0.17 released. It seems to be working well on my devices. Let me know if you see any issues.

**Comment by jole141** [comment]
@vitorpamplona Still not working on the iOS device. Getting "BLEAdvertiser.<function_name> is not a function." Please help :)

**Comment by maulanakurniawan** [comment]
For some reason, you need to add below entry into your Podfile manually to make it work. Put it inside your app target, then run pod install

`pod 'react-native-ble-advertiser', :path => '../node_modules/react-native-ble-advertiser/ios'`

**Comment by Kasendwa** [comment]
> For some reason, you need to add below entry into your Podfile manually to make it work. Put it inside your app target, then run pod install
> 
> `pod 'react-native-ble-advertiser', :path => '../node_modules/react-native-ble-advertiser/ios'`

Based off this fix highlighted by @maulanakurniawan , here is how you'd implement this in an expo managed project.

1. Create a file somewhere in your project. Ex: `custom-plugins/react-native-ble-advertiser.js` and paste the code below in it.

```
const { withDangerousMod } = require('@expo/config-plugins');
const { mergeContents } = require('@expo/config-plugins/build/utils/generateCode');
const fs = require('fs');
const path = require('path');

function withPodfile(config) {
	return withDangerousMod(config, [
		'ios',
		async config => {
			const filePath = path.join(config.modRequest.platformProjectRoot, 'Podfile');
			const contents = fs.readFileSync(filePath, 'utf-8');

			const addedPod = mergeContents({
				tag: 'Add the BLEAdvertiser pod',
				src: contents,
				newSrc: `pod 'react-native-ble-advertiser', :path => '../node_modules/react-native-ble-advertiser/ios'`,
				anchor: 'use_expo_modules!',
				offset: 0,
				comment: '#'
			});

			if (!addedPod.didMerge) {
				console.log(
					"ERROR: Cannot add BLEAdvertiser pod the project's ios/Podfile because it's malformed. Please report this with a copy of your project Podfile."
				);
				return config;
			}

			fs.writeFileSync(filePath, addedPod.contents);

		return config;
		}
	]);
}

module.exports = function withBLEAdvertiserPodFix(config) {
	config = withPodfile(config);
	return config;
};
```

Then in your `expo` config file, add the following under expo.plugins

```
{
    "expo": {
         "plugins": [
            ...,
            "./custom-plugins/react-native-ble-advertiser.js"
        ]
    }
}

**Comment by leodevbro** [comment]
@Kasendwa, please help me to setup Expo project with this package (**react-native-ble-advertiser**). Does your solution work also on Android? If not, could you please provide a minimal Expo project with this package working on both iOS and Android? It is really a big dream for me to make it possible with the simple flow of Expo (Without the need of Android Studio and Mac XCode).

Here:
https://github.com/vitorpamplona/react-native-ble-advertiser/pull/55
@fhllnd did a great job by providing the **bare** React Native working minimal project (_it is updated example project, because the old example project no longer works_), and it seems very useful and informative code even if I somehow manage to setup **react-native-ble-advertiser** with Expo.

I was going to go with bare React Native flow, but I discovered that nowadays Expo became too powerful to ignore. It [no longer needs ejecting](https://docs.expo.dev/workflow/customizing/) and it can just implement unsupported npm packages by [config plugins](https://docs.expo.dev/config-plugins/introduction/) (_like your solution_) and/or [development builds](https://docs.expo.dev/develop/development-builds/introduction/) and/or [creating native modules](https://docs.expo.dev/workflow/customizing/#creating-native-modules). The thing is that I am still very new to RN world and I feel I need a little help to implement this package as a config plugin (or something like that), just to get the minimal project setup.

**Comment by Kasendwa** [comment]
Hi @leodevbro, sorry I had missed this. Let me look into it tomorrow and get back to you.

**Comment by leodevbro** [comment]
> Hi @leodevbro, sorry I had missed this. Let me look into it tomorrow and get back to you. 

Oh, it's ok. Thanks. Sure, I can wait many days or many weeks. I'm just trying to at least find out if this is possible without a big headache.

**Comment by leodevbro** [comment]
@Kasendwa, Hi again, just wanted to ask, have you had any opportunity to have a look at it? I understand if you have limited time.
